lib/command.c

```diff
@@ -1395,6 +1395,21 @@ DEFUN (config_write,
   /* Get filename. */
   config_file = host.config;
 
+#ifndef O_DIRECTORY
+#define O_DIRECTORY 0
+#endif
+  slash = strrchr (config_file, '/');
+  if (slash)
+    {
+      char *config_dir = XSTRDUP (MTYPE_TMP, config_file);
+      config_dir[slash - config_file] = '\0';
+      dirfd = open(config_dir, O_DIRECTORY | O_RDONLY);
```

Comment:
Update to conversation in slack:

opendir returns a DIR *, fsync needs a fd.  So comment is silly.  IGNORE!